### PR TITLE
[WIP] Add EIP155 signature verification

### PR DESCRIPTION
**Pull request by annieke**
Addresses https://github.com/ethereum-optimism/roadmap/issues/612

currently having trouble serializing the transaction due to chainID and `sig.v` mismatch:

```
Error: transaction.chainId/signature.v mismatch (argument="signature", value={"r":"0x449600eb202f77605ce4fb6210cd0cb2a93b0831f00af3f3ee4266778728c9be","s":"0x590ebe4a5a275bc69b9ff190f57eebde0a88249e773e5c283e27aa9248b6d81b","v":0}, code=INVALID_ARGUMENT, version=transactions/5.0.10)
```

### CODE REVIEW

File: test/unit-tests/services/l1-ingestion/handlers/sequencer-batch-appended.spec.ts [hunk 1]
```diff
@@ -43,38 +43,49 @@ describe('Event Handlers: OVM_CanonicalTransactionChain.SequencerBatchAppended',
         expect(output1).to.equal(expected1)
       })
 
-      it('should mark a transaction as valid if the `v` parameter is 1', () => {
+      it('should mark a transaction as invalid if the `v` parameter is greater than 1', () => {
```
Comment (annieke) [inline test/unit-tests/services/l1-ingestion/handlers/sequencer-batch-appended.spec.ts:46]: i probably accidentally changed this while testing, will change it back

File: src/services/l1-ingestion/handlers/sequencer-batch-appended.ts [hunk 2]
```diff
@@ -279,12 +279,42 @@ export function validateBatchTransaction(
...
+    delete reformattedTx.target
+    const serializedTx = ethers.utils.serializeTransaction(
+      reformattedTx,
+      decoded.sig
+    )
+    const recoveredAddress = ethers.utils.recoverAddress(
+      // getting an error during this serialization from a chainID and signature.v mismatch
+      serializedTx,
+      decoded.sig
+    )
```
Comment (annieke) [inline src/services/l1-ingestion/handlers/sequencer-batch-appended.ts:305]: if i'm understanding this correctly, we will need to serialize the decoded tx. i'll try this tomorrow without the signature?

**Comment by tynes** [comment]
> currently having trouble serializing the transaction due to chainID and sig.v mismatch:
It sounds like its expecting the chainId to be encoded in the v parameter. The chain id is used to prevent replay attacks, meaning a transaction intended to be sent on one network is sent on another network. See https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md

We need the OVM ChainId which is possible to get through calling the Execution Manager Contract on L1 or the sequencer
See https://github.com/ethereum-optimism/contracts-v2/blob/f6069a881fbf6c35687a1676a73c67596b3ef4f9/contracts/optimistic-ethereum/iOVM/execution/iOVM_ExecutionManager.sol#L93

File: test/unit-tests/services/l1-ingestion/handlers/sequencer-batch-appended.spec.ts (hunk 1 above)
Comment (annieke) [reply test/unit-tests/services/l1-ingestion/handlers/sequencer-batch-appended.spec.ts:46]: wait this is right lol, since this won't be the validation now

**Comment by annieke** [comment]
currently this check is largel meaningless since both addresses are derived from the same signature, however we could compare the signature decoded in the `data` field:
https://github.com/ethereum-optimism/go-ethereum/blob/c2d0b423887770e423a917065683805139b26c1c/core/state_transition_ovm.go#L105-L118

**Comment by snario** [comment]
What's the status of this PR?

**Comment by snario** [comment]
- We have a few transactions on the L1 chain which are invalid
- We still need to ingest them because they're on the L1 chain
- The point of this PR was to see if there was a way to learn that they're invalid 
- We discovered it requires too much contextual information and so we decided against this approach

**Pull request closed without merging**